### ISSUE TITLE
[a11y]:  [Windows OS High Contrast Mode]: 'Select' drop down menu not picking up right color

### ISSUE THREAD

**Issue by nkn2022** [carbon-design-system/carbon#16299]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

Windows

### Package version

7.59.13

### React version

16.14.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

Our product is using "Select" and "Dropdown" components to server as selectors.
In high contrast mode, the "Select" component does not seem to pick up right color for the user defined high contrast mode. It shows white background when dark mode is selected. The "Dropdown" component seems to be working correctly. 
I checked with the `Accessibility@IBM` team and understand having such mixed modes fails the compatibility requirement for running in Windows OS High Contrast mode/theme. So this will need to reviewed and fixed.

Example(Select): https://v7-react.carbondesignsystem.com/?path=/story/components-select--default
<img width="483" alt="image" src="https://github.com/carbon-design-system/carbon/assets/102628718/343ed3be-c000-4187-938a-6a6a3ff1b027">

Example(Dropdown) - https://v7-react.carbondesignsystem.com/?path=/story/components-dropdown--playground
<img width="395" alt="image" src="https://github.com/carbon-design-system/carbon/assets/102628718/dc2c985d-ca64-4dd8-bf89-c1d8d906b7cc">

### WCAG 2.1 Violation

1.1.1 Non-text Content

### Reproduction/example

NA

### Steps to reproduce

Turn on High Contract Mode on Windows. 
Use the storybook links provided in the description above. 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by nkn2022** [comment]
Also an issue in Carbon 11 - https://react.carbondesignsystem.com/?path=/docs/components-select--overview
<img width="1057" alt="image" src="https://github.com/user-attachments/assets/c750b9e2-cb8e-455e-ae96-9a2b95dda25b">

**Comment by wkeese** [comment]
This works fine for me, both Carbon 10 and Carbon 11, except for a missing border:

<img width="686" alt="Image" src="https://github.com/user-attachments/assets/d5b094f3-32dd-4431-96ea-cc24d98f9403" />




> Turn on High Contract Mode on Windows.

You didn't say which version of Windows you are using, or which high contrast mode you selected.

I'm testing on Windows 11, Aquatic theme.

**Comment by nkn2022** [comment]
@wkeese Thanks for looking into this. I was using Windows 11 fire machine. Your screenshot looks correct to me. I will just retest on windows and confirm soon.
 
One question:
When I test on my Mac with our component UI; switching to dark mode the `Select` looks completely different from a `Dropdown`. Why is this?  Is this expected? There is no overriding of the colors from our side.

A `Select` component 

<img width="517" alt="Image" src="https://github.com/user-attachments/assets/e7d09d23-39a6-40fb-bd08-89cca51f4344" />

A `Dropdown` component

<img width="680" alt="Image" src="https://github.com/user-attachments/assets/03887ee2-db98-4be6-8bd3-3c4828ea230d" />

**Comment by wkeese** [comment]
I think that's just how Mac displays selects (or at least, Safari on Mac).  IIRC you can't override the styling.

It's sort-of like how the select behavior on an iPhone is based on the OS rather than any CSS in Carbon.

**Comment by nkn2022** [comment]
I tested the Carbon 11 on Windows high contract mode and it looks correct now. Not sure why it showed a white background before.

**Select**
<img width="805" alt="Image" src="https://github.com/user-attachments/assets/7c948906-6406-46c5-b4a9-71bc48b45667" />

**Dropdown**
<img width="429" alt="Image" src="https://github.com/user-attachments/assets/510d5c3e-eac0-4c47-8690-4f347fd7ad7c" />

As you mentioned the difference between the 2 components in dark mode is probably OS dependent, so I guess we can ignore this for now. The background showing white was the real issue I opened which is not seen anymore. May be I had some complex setting at the time. Anyways, I will go ahead and close this for now. 

Thanks a lot for checking!
